### PR TITLE
Use fp32 compute_type when calling rocBLAS API

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

File: src/targets/gpu/gemm_impl.cpp [hunk 1]
```diff
@@ -97,64 +104,124 @@ void gemm_impl(context& ctx,
...
+                               m * n,
+                               is_3inputs ? to_pointer(args[3]) : to_pointer(args[2]),
+                               output_type,
+                               ldc,
+                               m * n,
+                               num_matrices,
+                               compute_type,
+                               rocblas_gemm_algo_standard,
+                               0,
+                               flag);
```
Comment (pfultz2) [inline src/targets/gpu/gemm_impl.cpp:193]: Can we combine this function call instead doing two calls? Maybe we can create the alpha/beta pointers before calling the function, etc.

Comment (kahmed10) [reply src/targets/gpu/gemm_impl.cpp]: fixed

**Review by scxiao — APPROVED** [review]

File: src/targets/gpu/gemm_impl.cpp [hunk 2]
```diff
@@ -77,8 +83,20 @@ void gemm_impl(context& ctx,
...
     output_shape.visit_type([&](auto as) {
-        auto alpha_r    = as(alpha);
-        auto beta_r     = as(beta);
+
+        auto alpha_r = as(alpha);
+        auto beta_r  = as(beta);
+
+        // use void pointer to select different data type if using fp32 mode
+        void* alpha_v{&alpha_r};
+        void* beta_v{&beta_r};
```
Comment (pfultz2) [inline src/targets/gpu/gemm_impl.cpp:92]: Can do this with assignments instead?

```cpp
void* alpha_v = &alpha_r;
void* beta_v = &beta_r;
```

Comment (kahmed10) [reply src/targets/gpu/gemm_impl.cpp]: fixed

**Comment by kahmed10** [comment]
> @kahmed10 Looks like this change makes the bert model output to be incorrect.

The results will be different when using the fp32 compute_type, but after some initial testing on real test inputs, the results look okay.

With PR:
```
+ /src/rocm-migraphx/tools/migx/build/migx --glue=MRPC --gluefile=/src/rocm-migraphx/datasets/glue/MRPC.tst --tfpb /src/rocm-migraphx/saved-models/bert_mrpc1.pb
+ tee /src/rocm-migraphx/test-results/bertmrpc-2022-03-02-19-40/bertmrpc-fp32.out
accuracy rate = 0.862745
+ /src/rocm-migraphx/tools/migx/build/migx --glue=MRPC --gluefile=/src/rocm-migraphx/datasets/glue/MRPC.tst --tfpb /src/rocm-migraphx/saved-models/bert_mrpc1.pb --fp16
+ tee /src/rocm-migraphx/test-results/bertmrpc-2022-03-02-19-40/bertmrpc-fp16.out
accuracy rate = 0.865196
```

With develop:
```
+ /src/rocm-migraphx/tools/migx/build/migx --glue=MRPC --gluefile=/src/rocm-migraphx/datasets/glue/MRPC.tst --tfpb /src/rocm-migraphx/saved-models/bert_mrpc1.pb
+ tee /src/rocm-migraphx/test-results/bertmrpc-2022-03-02-19-55/bertmrpc-fp32.out
accuracy rate = 0.862745
+ /src/rocm-migraphx/tools/migx/build/migx --glue=MRPC --gluefile=/src/rocm-migraphx/datasets/glue/MRPC.tst --tfpb /src/rocm-migraphx/saved-models/bert_mrpc1.pb --fp16
+ tee /src/rocm-migraphx/test-results/bertmrpc-2022-03-02-19-55/bertmrpc-fp16.out
accuracy rate = 0.865196
```

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**